ui.frontend/src/customFunctions.js

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright 2022 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+/**
+ *
+ * @param str {string} json string to convert custom function to object
+ * @return {object} JSON Object after parsing the string as json. In case of
+ * exceptions empty object is returned
+ */
+function toObject(str) {
+    try {
+        return JSON.parse(str);
+    }
+    catch (e) {
+        return {}
+    }
+}
+
+/**
+ * prefixes the url with the context path
+ * @param url {string}
+ * @returns {string}
+ */
+function externalize(url) {
+    if (window?.Granite?.HTTP && typeof (window.Granite.HTTP.externalize === "function")) {
+        return window.Granite.HTTP.externalize(url);
+    } else {
+        return url;
+    }
+}
+
+export const customFunctions = {
```

Comment:
are these  custom functions really needed here? can we defined them in json-formula instead?

Reply:
externalize would need to be defined somewhere outside of json formula as it is aem specific. I also wanted these to be examples so that more functions can be registered.

Raised an issue in JSON Formula for toObject (or similar)

https://github.com/adobe/json-formula/issues/35